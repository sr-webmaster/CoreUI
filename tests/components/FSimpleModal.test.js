import { mount } from '@vue/test-utils'
import Component from '../../src/components/FSimpleModal.vue'

describe('SimpleModal', () => {
  const item = {
    title: 'Thank you for your request',
    text: 'Please stand by for admin validation'
  }
  it('snapshot default', async () => {
    const wrapper = mount(Component, {
      propsData: {
        title: item.title,
        text: item.text
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.text()).toContain(item.title)
  })
  test('click on close button', async () => {
    const wrapper = mount(Component)
    wrapper.find('button').trigger('click')
    expect(wrapper.emitted().ok).toHaveLength(1)
  })
})
