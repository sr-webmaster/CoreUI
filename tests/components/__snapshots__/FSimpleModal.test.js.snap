// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleModal snapshot default 1`] = `
<div
  class="pa-3 v-card v-sheet theme--light"
>
  <div
    class="v-card__title px-0 justify-center"
  >
    <div
      class="title font-weight-bold"
    >
      
      Thank you for your request
    
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="py-5 px-4 text-xs-center"
  >
    
    Please stand by for admin validation
  
  </div>
   
  <div
    class="layout row"
  >
    <div
      class="flex xs6 offset-xs3"
    >
      <div
        class="v-card__actions px-8"
      >
        <button
          class="v-btn v-btn--block theme--light primary"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
          Ok
        
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
